packages/shell-api/src/shell-internal-state.ts

```diff
@@ -169,6 +174,13 @@ export default class ShellInternalState {
       });
     }
 
+    contextObject.passwordPrompt = async(): Promise<string> => {
```

Comment:
Hi! I wasn't sure if you wanted input on this yet since its a WIP, apologies if it's too early!

I would consider adding this method to the top level `ShellApi` class (in shell-api.ts) if possible. It should then get automatically added to ContextObject on line 114/119. That way it would be as simple as using the `returnsPromise` decorator for the async rewriter to know it needs an await inserted before each use. The ShellApi object has access to the internal state object via `this._internalState` so the `evaluationListener` should still be accessible.

Reply:
If I'm understanding it all right, it should work pretty much the same as the `connect` method

Reply:
Oh nice, thank you, I think that’s exactly what I was looking for!

I think all this still needs are tests then. :slightly_smiling_face:

Reply:
Awesome, it looks good!